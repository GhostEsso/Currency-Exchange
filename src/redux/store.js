import { configureStore } from '@reduxjs/toolkit';
import exchangeSlice from './exchangeSlice';

const store = configureStore({
  reducer: {
    exchange: exchangeSlice.reducer,
  },
});

export default store;
