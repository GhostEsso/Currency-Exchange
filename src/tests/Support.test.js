import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Support from './Support';

import '@testing-library/jest-dom/extend-expect';

jest.mock('axios', () => ({
  get: jest.fn().mockResolvedValue({
    data: {
      supportedCurrenciesMap: {},
    },
  }),
}));

test('Support component renders correctly', () => {
  // Arrange: Render the Support component within the required context
  const { getByText, asFragment } = render(
    <Provider store={store}>
      <Router>
        <Support />
      </Router>
    </Provider>,
  );

  // Act: Attempt to find the heading element
  const headingElement = getByText('Currency not found');

  // Assert: Check if the heading element is present
  expect(headingElement).toBeInTheDocument();

  expect(asFragment()).toMatchSnapshot();
});
