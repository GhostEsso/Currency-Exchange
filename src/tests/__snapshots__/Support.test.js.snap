// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Support component renders correctly 1`] = `
<DocumentFragment>
  <section
    class="support"
  >
    <div
      class="details"
    >
      <h2>
        Currency not found
      </h2>
    </div>
  </section>
</DocumentFragment>
`;

exports[`renders Support component 1`] = `
<DocumentFragment>
  <section
    class="support"
  >
    <div
      class="details"
    >
      <h2>
        Currency not found
      </h2>
    </div>
  </section>
</DocumentFragment>
`;
